### PR TITLE
Publish Magento Cloud 2FA for SSH beta documentation

### PR DESCRIPTION
**Pull request by meker12**
## Purpose of this pull request

- Updated the Manage user access topic with security configuration information:
  - New Create API token and Enable 2FA sections
  - Streamlined the project and environment user role reference information
  - New "Adding additional authentication requirements" section for MFA enforcement feature 
  - Added screen captures for 2FA setup and enablement
  - Added screen captures for API token generation
- Added the Enable MFA enforcement for SSH topic with information about enabling
MFA enforcement on a Magento Cloud project.
- Added instructions for authenticating
on projects with MFA enforcement enabled.
## Affected DevDocs pages

- https://devdocs.magento.com/cloud/project/project-enable-mfa-project.html
- https://devdocs.magento.com/cloud/project/project-setup-split-db.html
- https://devdocs.magento.com/cloud/project/project-webint-basic.html
- https://devdocs.magento.com/cloud/project/projects.html
- https://devdocs.magento.com/cloud/project/user-admin.html
- New images
   - https://devdocs.magento.com/common/images/cloud-2fa-settings-tfa-recovery-codes.png
   - https://devdocs.magento.com/common/images/cloud/Cloud_Projects.png
   - https://devdocs.magento.com/common/images/cloud/cloud-2fa-settings-tfa-app-example.png
  - https://devdocs.magento.com/common/images/cloud/cloud-2fa-settings-tfa-browser-trust.png
  - https://devdocs.magento.com/common/images/cloud/cloud-2fa-settings-tfa-manage.png
  - https://devdocs.magento.com/common/images/cloud/cloud-2fa-settings-tfa-recovery-codes.png
  - https://devdocs.magento.com/common/images/cloud/cloud-account-settings-manage-2fa-config.png
  - https://devdocs.magento.com/common/images/cloud/cloud-account-settings-tab.png
  - https://devdocs.magento.com/common/images/cloud/cloud_account_settings-api-token-app-name.png
  - https://devdocs.magento.com/common/images/cloud/cloud_account_settings-api-token-created.png
  - https://devdocs.magento.com/common/images/cloud/cloud_account_settings-create-api-token.png
  - https://devdocs.magento.com/common/images/cloud/cloud_account_settings-tfa-setup-app.png
  - https://devdocs.magento.com/common/images/cloud/cloud_account_settings-tfa-setup-steps.png
  - https://devdocs.magento.com/common/images/cloud_account_settings-enable-tfa.png

whatsnew
Updated _Cloud Guide_ with information about account and project security options:
- Added instructions to configure 2FA security and API tokens setting to the [Manage user access](https://devdocs.magento.com/cloud/project/user-admin.html) topic.
- Added [Enable MFA enforcement for SSH](https://devdocs.magento.com/cloud/project/project-enable-mfa-project.html) topic to describe this project-level feature to enforce 2FA requirement for SSH access to Magento Cloud environments. Also added instructions for authenticating on projects that have MFA enforcement enabled.

### CODE REVIEW

File: src/cloud/project/user-admin.md [hunk 1]
```diff
@@ -176,8 +164,118 @@ Add an environment-level user:
...
+
+1. On the Account settings tab, expand the **Security** section. Then, click **Set up application**.
+
+   ![Cloud Security settings]({{ site.baseurl }}/common/images/cloud/
+   cloud_account_settings-tfa-setup-app.png){:width="550px"}
+
+1. Use the linked instructions to install one of the approved authentication applications on your mobile device, if needed.
+
+   ![Cloud Security settings]({{ site.baseurl }}/common/images/cloud/
+   cloud_account_settings-tfa-setup-app-options.png){:width="550px"}
```
Comment (hguthrie) [inline src/cloud/project/user-admin.md:214]:
```suggestion
   ![Cloud Security settings]({{ site.baseurl }}/common/images/cloud/cloud_account_settings-tfa-setup-app-options.png){:width="550px"}
```

File: src/cloud/project/user-admin.md [hunk 2]
```diff
@@ -176,8 +164,118 @@ Add an environment-level user:
...
+1. Log in to the [{{site.data.var.ece}} user account][{{site.data.var.ece}} account].
+
+1. On the Cloud projects page, click the **Account settings** tab.
+
+   ![Cloud projects page]({{ site.baseurl }}/common/images/cloud/cloud-account-settings-tab.png){:width="550px"}
+
+1. On the Account settings tab, expand the **Security** section. Then, click **Set up application**.
+
+   ![Cloud Security settings]({{ site.baseurl }}/common/images/cloud/
+   cloud_account_settings-tfa-setup-app.png){:width="550px"}
```
Comment (hguthrie) [inline src/cloud/project/user-admin.md:209]:
```suggestion
   ![Cloud Security settings]({{ site.baseurl }}/common/images/cloud/cloud_account_settings-tfa-setup-app.png){:width="550px"}
```

File: src/cloud/project/user-admin.md (hunk 1 above)
Comment (meker12) [reply src/cloud/project/user-admin.md]: Thanks for the suggestions-- I didn't see them before building and pushing from local. Build running now 🤞

File: src/cloud/project/project-enable-mfa-project.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
+---
+group: cloud-guide
+title: Enable MFA enforcement for SSH access
+functional_areas:
+  - Cloud
+  - Configuration
+---
+
+For added security, {{site.data.var.ece}} provides an mutlitfactor authentication (MFA) enforcement feature to manage multifactor authentication requirements for SSH access to Cloud infrastructure.
```
Comment (meker12) [inline src/cloud/project/project-enable-mfa-project.md:9]:
```suggestion
For added security, {{site.data.var.ece}} provides a mutlitfactor authentication (MFA) enforcement feature to manage multifactor authentication requirements for SSH access to Cloud infrastructure.
```

File: src/cloud/project/project-enable-mfa-project.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+When this feature is enabled on a project, any user or machine account establishing an SSH connection to an environment must follow an authentication workflow that requires a two-factor authentication (2FA) code and an SSH certificate, or an API token to access the environment.
+
+{:bs.callout-important}
```
Comment (meker12) [inline src/cloud/project/project-enable-mfa-project.md:13]:
```suggestion
{:.bs.callout-important}
```

File: src/cloud/project/project-enable-mfa-project.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+The MFA enforcement feature is not enabled by default on {{site.data.var.ece}} projects. You must submit a Magento support ticket to request 2FA enablement on your project. After enablement, all subsequent SSH connection requests require either a 2FA token or an API token.
+
+## Certificates for SSH access
+
+The MFA enforcement feature allows users to exchange an OAUTH access token with a short-lived SSH certificate generated by the Magento Cloud Certifier API.
+
+If the user accessing the environment has project Admin or contributor access, a valid SSH key, and a valid 2FA code or API token, {{site.data.var.ece}} uses these credentials to generate the temporary SSH certificate. The certificate expiration is set to 1 hour, but it refreshes automatically during the current session.
```
Comment (meker12) [inline src/cloud/project/project-enable-mfa-project.md:20]:
```suggestion
If the user accessing the environment has project Admin or contributor access, a valid SSH key, and a valid 2FA code, {{site.data.var.ece}} uses these credentials to generate the temporary SSH certificate. The certificate expiration is set to 1 hour, but it refreshes automatically during the current session.
```

File: src/cloud/project/project-enable-mfa-project.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+After logging into a project with MFA enforcement, users must run the following command to generate the certificate:
+
+```bash
+magento-cloud ssh-cert:load
+```
+
+This command generates the SSH certificate and installs the certificate in the SSH agent of the local user.
```
Comment (meker12) [inline src/cloud/project/project-enable-mfa-project.md:28]:
```suggestion
This command generates the SSH certificate and installs it in the SSH agent of the local user.
```

File: src/cloud/project/user-admin.md [hunk 7]
```diff
@@ -1,57 +1,45 @@
...
 
-You can assign the following project-level roles to users:
+## Add user authentication requirements
 
--  The **Super user** role grants administrator access to all environments. They can change settings and execute actions on any environment, including creating and restoring [snapshots][].
+For added security, Magento is introducing a project-level MFA enforcement feature to improve security for SSH access to Cloud infrastructure. See [MFA enforcement for SSH].
 
--  The **Project reader** role grants view access to all environments in a project. Users with this role cannot execute actions on any environment. However, you can update the configuration on each environment to add environment-level permission for users with Project reader access.
+When MFA enforcement is enabled on a {{site.data.var.ece}} project, all users that with SSH access to an environment in that project must enable 2FA on their {{site.data.var.ece}} account. For automated processes, users must create an API token that machine users can use to authenticate from the command line. See [Enable user accounts for 2FA and SSH access](#update-account-security-settings).
```
Comment (meker12) [inline src/cloud/project/user-admin.md:29]:
```suggestion
When MFA enforcement is enabled on a {{site.data.var.ece}} project, all users with SSH access to an environment in that project must enable 2FA on their {{site.data.var.ece}} account. For automated processes, users must create an API token that machine users can use to authenticate. See [Enable user accounts for 2FA and SSH access](#update-account-security-settings).
```

File: src/cloud/project/user-admin.md [hunk 8]
```diff
@@ -176,8 +164,116 @@ Add an environment-level user:
 
 {%include cloud/note-prevent-site-availability-issues.md%}
 
+## Update account security settings
+
+After you add a user to a Cloud project, ask the user to review their account settings and add the following security configuration as needed:
```
Comment (meker12) [inline src/cloud/project/user-admin.md:169]:
```suggestion
After you add a user to a Cloud project, ask them to review their account settings and add the following security configuration as needed:
```

File: src/cloud/project/user-admin.md [hunk 9]
```diff
@@ -176,8 +164,116 @@ Add an environment-level user:
...
+
+-  Enable two-factor authentication (TFA)
+
+   Magento recommends adding two-factor authentication to all accounts to meet security and compliance standards. Some projects will soon require two-factor authentication for all accounts that use SSH to connect to {{site.data.var.ece}} projects.
```
Comment (meker12) [inline src/cloud/project/user-admin.md:173]:
```suggestion
   Magento recommends adding two-factor authentication to all accounts to meet security and compliance standards. Some projects will soon require two-factor authentication on all accounts that use SSH to connect to {{site.data.var.ece}} projects.
```

File: src/cloud/project/user-admin.md [hunk 10]
```diff
@@ -176,8 +164,116 @@ Add an environment-level user:
...
+
+-  Enable SSH keys
+
+   Users that require access to {{site.data.var.ece}} source code repositories and infrastructure must enable SSH keys on their account. See [Enable SSH keys][].
+
+-  Create an API token
+
+   You can generate an API token on your account that can be used for secure machine access to an environment through an SSH connection. You can use the API token to enable login workflows for automated processes.
```
Comment (meker12) [inline src/cloud/project/user-admin.md:181]:
```suggestion
   You can generate an API token on your account that can be used for secure machine access to an environment through an SSH connection. Use the API token to enable login workflows for automated processes.
```

File: src/cloud/project/user-admin.md [hunk 11]
```diff
@@ -176,8 +164,116 @@ Add an environment-level user:
...
+1. Optional. Enable _Trusted browser_ settings to save current authentication code in the browser session for 30 days and reduce number of authentication challenges during project log in.
+
+1. Click **Save** or **Skip**.
+
+1. On the _TFA setup - Recovery_ codes page, copy and save the recovery codes so that you can log into your {{site.data.var.ece}} project when you cannot access your mobile device or authentication app.
+
+   ![Cloud TFA recovery codes]({{ site.baseurl }}/common/images/cloud/cloud-2fa-settings-tfa-recovery-codes.png){:width="550px"}
+
+   {:.bs.callout-warning}
+   Copy the recovery codes to another location or write them down to use if you lose access to your device or authentication app. Save the codes to your account to view and manage them from your account security settings. If you lose account access on a 2FA-enabled account and have no recovery codes, you must contact your Project administrator, or submit a Magento support ticket to reset the TFA application.
```
Comment (meker12) [inline src/cloud/project/user-admin.md:235]:
```suggestion
   Copy the recovery codes to another location or write them down to recover access if you your device or access to the authenticator application. Save the codes to your account to view and manage them from your account security settings. If you lose account access on a 2FA-enabled account and have no recovery codes, you must contact your Project administrator, or submit a Magento support ticket to reset the TFA application.
```

File: src/cloud/project/project-enable-mfa-project.md [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+group: cloud-guide
+title: Enable MFA enforcement for SSH access
+functional_areas:
+  - Cloud
+  - Configuration
+---
+
+For added security, {{site.data.var.ece}} provides a mutlitfactor authentication (MFA) enforcement feature to manage multifactor authentication requirements for SSH access to Cloud infrastructure.
+
+When this feature is enabled on a project, any user or machine account establishing an SSH connection to an environment must follow an authentication workflow that requires a two-factor authentication (2FA) code and an SSH certificate, or an API token to access the environment.
```
Comment (hguthrie) [inline src/cloud/project/project-enable-mfa-project.md:11]:
```suggestion
When MFA enforcement is enabled on a project, any user or machine account establishing an SSH connection to an environment must follow an authentication workflow that requires a two-factor authentication (2FA) code and an SSH certificate, or an API token to access the environment.
```

File: src/cloud/project/project-enable-mfa-project.md [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+
+{:.bs.callout-important}
+The MFA enforcement feature is not enabled by default on {{site.data.var.ece}} projects. You must submit a Magento support ticket to request 2FA enablement on your project. After enablement, all subsequent SSH connection requests require either a 2FA token or an API token.
+
+## Certificates for SSH access
+
+The MFA enforcement feature allows users to exchange an OAUTH access token with a short-lived SSH certificate generated by the Magento Cloud Certifier API.
+
+If the user accessing the environment has project Admin or contributor access, a valid SSH key, and a valid 2FA code, {{site.data.var.ece}} uses these credentials to generate the temporary SSH certificate. The certificate expiration is set to 1 hour, but it refreshes automatically during the current session.
```
Comment (hguthrie) [inline src/cloud/project/project-enable-mfa-project.md:20]: Admin is in caps, but not contributor.

File: src/cloud/project/project-enable-mfa-project.md [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+
+After logging into a project with MFA enforcement, users must run the following command to generate the certificate:
```
Comment (hguthrie) [inline src/cloud/project/project-enable-mfa-project.md:22]:
```suggestion
After logging into a project with MFA enforcement, users must use the CLI to generate the SSH certificate:
```

File: src/cloud/project/project-enable-mfa-project.md [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+
+```bash
+magento-cloud ssh-cert:load
+```
+
+This command generates the SSH certificate and installs it in the SSH agent of the local user.
```
Comment (hguthrie) [inline src/cloud/project/project-enable-mfa-project.md:28]:
```suggestion
The `ssh-cert:load` command generates the SSH certificate and installs it in the SSH agent of the local user.
```

File: src/cloud/project/project-enable-mfa-project.md [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+-  [Admin or contributor access to the {{site.data.var.ece}} environment][Manage user access]
+-  [SSH access key configured on account][add public SSH key]
+-  [2FA enabled on account][]
+
+{:.procedure}
+To connect using SSH with 2FA user account credentials:
+
+1. Log in to your {{site.data.var.ece}} account and authenticate using 2FA.
+
+1. After logging in, open a command line and run the following command to generate the certificate.
```
Comment (hguthrie) [inline src/cloud/project/project-enable-mfa-project.md:47]: open a command line?
run the command is used too much.

```suggestion
1. On your local workstation, use the CLI to generate an SSH certificate.
```

File: src/cloud/project/project-enable-mfa-project.md [hunk 17]
```diff
@@ -0,0 +1,127 @@
...
+     Expires at: 2020-07-13T15:28:13-04:00
+     Multi-factor authentication: verified
+     Mode: interactive
+   The certificate will be automatically refreshed when necessary.
+   Checking SSH configuration file: /Users/<user-name>/.ssh/config
+   Do you want to update the file automatically? [Y/n] Y
+   Configuration file updated successfully: /Users/<user-name>/.ssh/config
+   ```
+
+1. Connect to the project using SSH:
```
Comment (hguthrie) [inline src/cloud/project/project-enable-mfa-project.md:64]: Must this be master? Probably need to be specific.

File: src/cloud/project/project-enable-mfa-project.md (hunk 13 above)
Comment (meker12) [inline src/cloud/project/project-enable-mfa-project.md:20]:
```suggestion
If the user accessing the environment has project Admin or Contributor access, a valid SSH key, and a valid 2FA code, {{site.data.var.ece}} uses these credentials to generate the temporary SSH certificate. The certificate expiration is set to 1 hour, but it refreshes automatically during the current session.
```

File: src/cloud/project/project-enable-mfa-project.md (hunk 17 above)
Comment (meker12) [inline src/cloud/project/project-enable-mfa-project.md:64]:
```suggestion
1. Connect to the Cloud environment using SSH:
```

**Comment by meker12** [comment]
running tests

**Comment by meker12** [comment]
running tests

**Comment by meker12** [comment]
running tests

**Comment by meker12** [comment]
running tests

**Pull request merged**